### PR TITLE
Break out quoted refs check and add fix

### PR DESCRIPTION
**Pull request by danzilio**
This PR breaks out the `resource_reference_with_unquoted_title` check and adds a `fix` method.

### CODE REVIEW

**Comment by tuxmea** [comment]
Travis error on Ruby 1.9.3 only. Looks good to me.
We should remove ruby 1.9 from travis testing, too.

**Comment by ekohl** [comment]
>  We should remove ruby 1.9 from travis testing, too.

:+1: for that. I'll happily merge a PR for that.

**Comment by tuxmea** [comment]
@danzilio Many thanks for the PR.
Can you please rebase? I will merge afterwards.

File: lib/puppet-lint/plugins/resource_reference_with_unquoted_title.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          end
+        end
+      end
+    end
+  end
+
+  def fix(problem)
+    problem[:token].type = :SSTRING
+  end
+end
```
Comment (bastelfreak) [inline lib/puppet-lint/plugins/resource_reference_with_unquoted_title.rb:39]: please fix the lineending here :)

File: spec/puppet-lint/plugins/resource_reference_with_unquoted_title_spec.rb [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      it 'should fix the manifest' do
+        expect(problems).to contain_fixed(msg).on_line(3).in_column(31)
+      end
+
+      it 'should single quote the resource title' do
+        expect(manifest).to eq(fixed)
+      end
+    end
+  end
+end
```
Comment (bastelfreak) [inline spec/puppet-lint/plugins/resource_reference_with_unquoted_title_spec.rb:207]: and here :)

**Comment by tuxmea** [comment]
@danzilio any updates on this PR? have you seen @bastelfreak comments?

**Pull request merged**